### ISSUE TITLE
[Bug] Python doctests must be updated for numpy 2.0.1 

### ISSUE THREAD

**Issue by echoix** [OSGeo/grass#4100]
## Describe the bug

Doctest running with the gunittest test suite fail with numpy 2.0.1, as the datatype (like int32) doesn't appear when representing the shape of the array.

I don't know how to support both older numpy and numpy 2.0.1+ on the same doctest, as they don't have the same string representation, but it doesn't really bother, as it was aliases to the python types for a long while.

I observed these bugs last week on some Windows CI runs, as the package managers have recent versions quite fast, but didn't catch on that it was numpy.

## To reproduce

1. Install numpy 2 through pip
2. Run gunittest for `File "etc/python/grass/pygrass/utils.py", line 278, in grass.pygrass.utils.get_raster_for_points`, or on Windows for: `File "etc\python\grass\pygrass\raster\__init__.py", line 412, in grass.pygrass.raster.RasterSegment.put_row`
4. See error in output

## Screenshots

Windows:
https://github.com/echoix/grass/actions/runs/10085495709/job/27886383708?pr=168#step:11:1767 (one of my runs)
![image](https://github.com/user-attachments/assets/e53a453d-3e31-4a5c-b120-b0753e3a8719)
https://github.com/OSGeo/grass/actions/runs/10075548005/job/27854053510#step:11:1603 (the last commit on main)
![image](https://github.com/user-attachments/assets/78a2bdbe-03bb-4741-9b8e-55e6dd7c6fe0)

Ubuntu:
https://github.com/echoix/grass/actions/runs/10085634573/job/27886769880?pr=174#step:19:88 (my branch, with Python 3.12 and numpy 2.0.1 running gunittest+code coverage, but the exact commit might change as I'm rebasing my work for a PR).
![image](https://github.com/user-attachments/assets/459e55fa-771e-4b23-9e88-7b2c1720d221)

## System description
- Operating System: Windows (Windows Server 2019 on GitHub Actions CI) or Linux (Ubuntu 22.04, on GitHub Actions CI)
- GRASS GIS version: main at commit 0306f9333d0ce45e32ee970af569e876c99918aa, but in reality anything since numpy 2 could be installed by pip.
- Ubuntu 22.04 on Github Actions gives numpy 1.21.5-1 with apt, from python3-numpy  [`Get:155 http://azure.archive.ubuntu.com/ubuntu jammy-updates/main amd64 python3-numpy amd64 1:1.21.5-1ubuntu22.04.1 [3467 kB]`](https://github.com/OSGeo/grass/actions/runs/10075548025/job/27854054427#step:5:318)

## Additional context

I did a small simulation in a branch based on one of my pending work, using Python 3.12 for Ubuntu gunittest, and requires installing dependencies through pip rather than using the system's package manager.

**Comment by echoix** [comment]
As I finished writing, I understood that on Windows, we still only have 1.26.4-1 from OSGeo4W:
![image](https://github.com/user-attachments/assets/1692d8d3-2291-4c2f-956d-3ecd938b311f)

So it's not purely a numpy 2+ issue.

**Comment by Khansa435** [comment]
Hi @echoix, I can see there's a merged PR related to this issue. Is this fully resolved, or are there still remaining cases that need to be fixed?

I'm preparing for GSoC 2026 and looking to contribute to GRASS GIS. If there's still work to be done here, I'd be happy to take it up. Any guidance on what's left would be greatly appreciated :)

**Comment by echoix** [comment]
No, if you read the merged PR, you'll see my message that explains the warnings that were visible, and linked to this issue and others, and other solution hints.

The problem is still there.

Since the time this issue was filed, we now use a newer Ubuntu version (on half the Ubuntu CI jobs running gunittest). If there, a newer version of numpy is used, maybe we could simply adapt the doctests to use the new representation instead of the old one.

On second thought, since we recently branched off 8.5 and the main branch is early in 8.6, and a new Ubuntu 26.04 will be available at some point during the development, it is a good time to bump up the numpy version for the tests (having the complete doctests assume numpy 2 string representation).

You might also want to make sure some of the failing tests because of that failing representation weren't excluded. If so, they should be reenabled.

**Comment by Khansa435** [comment]
Hi @echoix , I'm sorry for the delay on this due to health issues :/
I have just opened a PR #7194  applying the proposed changes. Please take a look :)
